File: woodroid/src/main/java/me/gilo/woodroid/repo/product/CategoryRepository.java
package me.gilo.woodroid.repo.product;

import me.gilo.woodroid.data.api.CustomerAPI;
import me.gilo.woodroid.data.api.ProductCategoryAPI;
import me.gilo.woodroid.models.Category;
import me.gilo.woodroid.models.Customer;
import me.gilo.woodroid.repo.WooRepository;
import retrofit2.Call;

import java.util.List;

public class CategoryRepository extends WooRepository {

    private final ProductCategoryAPI apiService;

    public CategoryRepository(String baseUrl, String consumerKey, String consumerSecret) {
       super("products/categories", baseUrl, consumerKey, consumerSecret);
       apiService = retrofit.create(ProductCategoryAPI.class);
    }

    public Call<Category> create(Category category) {
        return apiService.create(category);
    }


    public Call<Category> category(int id) {
        return apiService.view(id);
    }

    public Call<List<Category>> categories() {
        return apiService.list();
    }

    public Call<Category> update(int id, Category category) {
        return apiService.update(id, category);
    }

    public Call<Category> delete(int id) {
        return apiService.delete(id);
    }

    public Call<Category> delete(int id, boolean force) {
        return apiService.delete(id, force);
    }


}

File: woodroid/src/main/java/me/gilo/woodroid/repo/product/CategoryRepository.kt
package me.gilo.woodroid.repo.product

import me.gilo.woodroid.data.api.ProductCategoryAPI
import me.gilo.woodroid.models.Category
import me.gilo.woodroid.models.filters.ProductCategoryFilter
import me.gilo.woodroid.repo.WooRepository
import retrofit2.Call

class CategoryRepository(baseUrl: String, consumerKey: String, consumerSecret: String) :
    WooRepository(baseUrl, consumerKey, consumerSecret) {

    private val apiService: ProductCategoryAPI

    init {
        apiService = retrofit.create(ProductCategoryAPI::class.java)
    }

    fun create(category: Category): Call<Category> {
        return apiService.create(category)
    }


    fun category(id: Int): Call<Category> {
        return apiService.view(id)
    }

    fun categories(): Call<List<Category>> {
        return apiService.list()
    }

    fun categories(productCategoryFilter: ProductCategoryFilter): Call<List<Category>> {
        return apiService.filter(productCategoryFilter.filters)
    }

    fun update(id: Int, category: Category): Call<Category> {
        return apiService.update(id, category)
    }

    fun delete(id: Int): Call<Category> {
        return apiService.delete(id)
    }

    fun delete(id: Int, force: Boolean): Call<Category> {
        return apiService.delete(id, force)
    }


}
